fulp_modules/main_features/bloodsuckers/code/bloodsucker/bloodsucker_crypt.dm

```diff
@@ -602,6 +613,100 @@
 			return
 
 
+/obj/structure/bloodsucker/vassalrack/proc/shapeshift_victim(mob/living/carbon/human/user, mob/living/carbon/human/target)
+	var/datum/antagonist/bloodsucker/bloodsuckerdatum = user.mind.has_antag_datum(/datum/antagonist/bloodsucker)
+
+	if(target.stat == DEAD)
+		to_chat(user, "<span class='notice'>Do you wish to rebuild this body as something greater? This will remove any restraints they might have, make them into a mute husk, and will cost 150 Blood!</span>")
+		var/list/revive_options = list(
+			"Yes" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_yes"),
+			"No" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_no")
+			)
+		var/revive_response = show_radial_menu(user, src, revive_options, radius = 36, require_near = TRUE)
+		switch(revive_response)
+			if("Yes")
+				to_chat(user, "<span class='danger'>You have brought [target] back from the dead!</span>")
+				to_chat(target, "<span class='announce'>As blood drips over your body, your heart begins to beat... You live again!</span>")
+				user.blood_volume -= 150
+				target.revive(full_heal = TRUE, admin_revive = TRUE)
+				ADD_TRAIT(target, TRAIT_MUTE, BLOODSUCKER_TRAIT)
+				target.become_husk()
+				return
+			else
+				to_chat(user, "<span class='danger'>You decide not to revive [target].</span>")
+				unbuckle_mob(target)
+				useLock = FALSE
+				return
+
+	var/static/list/monster_choices = list(
+		TZIMISCE_HUSK,
+		TZIMISCE_TWOARMED,
+		TZIMISCE_CLAWMONSTER,
+		TZIMISCE_TRIPLECHESTED,
+	)
+	var/list/options = list()
+	options = monster_choices
+	var/answer = tgui_input_list(user, "We have the chance to shapeshift our victim into something greater, how should we mutilate their corpse?", "What do we do with our victim?", options)
+
+	var/make_monster = TRUE /// Are we making them into a monster?
+	var/mob/living/simple_animal/monster /// The monster the target is being transferred to upon succesful ritual.
+	var/blood_lost /// The amount of blood lost by the Tzimisce upon succesfully shapeshifting the target.
+	var/blood_gained /// The amount of blood gained by the Tzimisce upon a successful shapeshift.
+	var/lose_items = TRUE /// Do we lose items upon succesful rituals? Only FALSE for husks as they do not become simplemobs.
+	var/user_message = "You have shaped [target] into a two-armed monster!" /// The message that the Tzimisce gets upon a successful ritual.
+	var/target_message = "You've been turned into a monster!" /// The message that the victim gets upon a succesful ritual.
+
+	if(!answer)
+		to_chat(user, "<span class='notice'>You decide to leave your victim just the way they are.</span>")
+		return
+
+	switch(answer)
+		if(TZIMISCE_HUSK)
+			blood_gained = 200
```

Comment:
can you repeatedly husk a victim to farm blood from them? I don't see anything that'd prevent any of these transformations from being applied to someone who's already been husked.

Reply:
to_chat(user, "<span class='notice'>You've already shapeshifted [C]!</span>")

You cant shapeshift twice

Reply:
oh, I guess there's the vassalization check